modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/management_bar/components/filters/CheckboxesFilter.js

```diff
@@ -17,10 +17,22 @@ import {ClayCheckbox} from '@clayui/form';
 import PropTypes from 'prop-types';
 import React, {useState} from 'react';
 
-import getAppContext from '../Context';
+function getOdataString(value, key) {
+	if (!value || !value.length) {
+		return null;
+	}
+
+	return `${key}/any(x:${value
+		.map(
+			(v) =>
+				`(x eq ${
+					typeof v.value === 'string' ? `'${v.value}'` : v.value
+				})`
+		)
+		.join(' or ')})`;
+}
```

Comment:
This looks so like the `getOdataString()` in AutocompleteFilter.js that I wonder whether we should move *all* of these into a single file and share the bits that can be shared. I'm thinking low-level primitives that the various different `getOdataString()` implementations in each file then end up calling/composing. 🤷‍♂️

Reply:
Yeah, that makes sense. And in fact if you see the [diff](https://github.com/marco-leo/liferay-portal-ee/pull/968/files#diff-d4a0f694dca408eef628d47e32d95a8c) for COMMERCE-4091 they had it in a single file previously and moved towards this spread approach. I didn't want to make too many changes on the act of bringing this to master because I'm expecting some more changes in the very near future and I didn't want the code to diverge too much. But I will ping Fabio about this to get a sense of the idea behind spreading this instead of concentrating and reusing.